fetch_repositories.rb

```diff
@@ -0,0 +1,37 @@
+#!/usr/bin/env ruby
+
+require "yaml"
+require "json"
+require "net/http"
+
+def get_repos
+  repos = []
+  repos_content = YAML::load_file 'repos.yml'
+  repos_content['repos'].each do |url|
+    repo_name = /^.*\/(.*\/.*)$/.match(url)[1]
+    api_uri = URI("https://api.github.com/repos/#{repo_name}")
+    api_response =  Net::HTTP.get_response(api_uri)
+    unless api_response.code == "200"
+      puts "WARNING: #{api_uri} returned #{api_response.code}"
+      next
+    end
+    repos << JSON.parse(api_response.body)
+  end
+  repos
+end
+
+def write_repos(repos)
+  repos.each do |repo|
+    repo['name'] = repo['full_name'].gsub('/', '-')
```

Comment:
I don't think there are any existing cases but this could create a collision if, e.g. we had projects called `ros-infrastructure/repo` and `ros-infrastructure/repo`. There's only marginally less risk if we use a single underscore as a separator so let's try two dashes.

```suggestion
    repo['name'] = repo['full_name'].gsub('/', '--')
```

Reply:
I applied the suggestion and also updated the existing repos content. Check 9d0eef7